### PR TITLE
Map more Windows System errors

### LINKED ISSUE #2893: `<system_error>`: Several Windows system errors are not mapped
**Describe the bug**
Several Windows system error codes are not mapped in syserror.cpp

https://github.com/microsoft/STL/blob/main/stl/src/syserror.cpp#L27

The description for each of the errors are as follows:
`ERROR_FILENAME_EXCED_RANGE` -> The filename or extension is too long.
`ERROR_BAD_NET_NAME` -> The network name cannot be found.
`ERROR_NO_MORE_FILES` -> There are no more files.

**Command-line test case**
```
std::errc::filename_too_long == std::error_code{ERROR_FILENAME_EXCED_RANGE, std::system_category()}; // false
std::errc::no_such_file_or_directory == std::error_code{ERROR_BAD_NET_NAME, std::system_category()}; // false
std::errc::no_such_file_or_directory == std::error_code{ERROR_NO_MORE_FILES, std::system_category()}; // false
```

**Expected behavior**
`ERROR_FILENAME_EXCED_RANGE` should be mapped to `std::errc::filename_too_long` (ENAMETOOLONG)
`ERROR_BAD_NET_NAME` and `ERROR_NO_MORE_FILES` should be mapped to `std::errc::no_such_file_or_directory` (ENOENT)

**STL version**
The latest commit manifests this behavior

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #2893

[System Error Codes (0-499)](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-)

* **ERROR_BAD_NET_NAME**
  * 67 (0x43)
  * The network name cannot be found.

* **ERROR_FILENAME_EXCED_RANGE**
  * 206 (0xCE)
  * The filename or extension is too long.

### CODE REVIEW

**Comment by Morph1984** [comment]
Thank you for the quick fix!

File: stl/src/syserror.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@ namespace {
         {ERROR_INVALID_DRIVE, errc::no_such_device},
         {ERROR_CURRENT_DIRECTORY, errc::permission_denied},
         {ERROR_NOT_SAME_DEVICE, errc::cross_device_link},
+        {ERROR_NO_MORE_FILES, errc::no_such_file_or_directory},
```
Comment (strega-nil-ms) [inline stl/src/syserror.cpp:41]: Is this not `errc::too_many_files_open`?

Comment (Morph1984) [reply stl/src/syserror.cpp:41]: `ERROR_NO_MORE_FILES` is returned by a file iterator, for example, Windows returns this error when there are no more files in the call to `FindNextFileA` https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findnextfilea

Comment (strega-nil-ms) [reply stl/src/syserror.cpp:41]: I could not get this error after more than 100'000 files open, so maybe not.

Comment (strega-nil-ms) [reply stl/src/syserror.cpp:41]: I don't like this then; it's _not_ an error, it's certainly not `no_such_file_or_directory`

**Comment by strega-nil-ms** [comment]
Thanks!

**Review by strega-nil-ms — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving `<system_error>`'s behavior! :tada: :chart_with_upwards_trend: :smile_cat:

**Pull request merged**